libs/util/Predicates.arucas

```diff
@@ -0,0 +1,47 @@
+class Predicate{
+    static fun all(predicates...){
+        return fun (value){
+            foreach(pred : predicates){
+                if(!pred(value)){
+                    return false;
+                }
+            }
+            return true;
+        };
+    }
+    static fun any(predicates...){
+        return fun (value){
+            foreach(pred : predicates){
+                if(pred(value)){
+                    return true;
+                }
+            }
+            return false;
+        };
+    }
+    static fun and(a, b){
+        return fun (value){
+            return a(value) && b(value);
+        };
+    }
+    static fun or(a,b){
+        return fun (value){
+            return a(value) || b(value);
+        };
+    }
+    static fun xor(a,b){
+        return fun (value){
+            return a(value) ~ b(value);
+        };
+    }
+    static fun always_true(){
+        return fun (value){
+            return true;
+        };
+    }
+    static fun always_false(){
+        return fun (value){
```

Comment:
Naming `always_false` and `always_true`, camel case.
Also these lambdas should be global constants, there is no reason they should be evaluated every single time this function is ran.